src/Cloud/weglide/UploadFlight.cpp

```diff
@@ -58,27 +59,43 @@ MakeUploadFlightMime(CURL *easy, const WeGlideSettings &settings,
 }
 
 Co::Task<boost::json::value>
-UploadFlight(CurlGlobal &curl, const WeGlideSettings &settings,
+UploadFlight(CurlGlobal& curl, const char *url,
+             const struct Pilot &pilot,
              uint_least32_t glider_type,
              Path igc_path,
              ProgressListener &progress)
 {
-  NarrowString<0x200> url(settings.default_url);
-  url += "/igcfile";
-
   CurlEasy easy{url};
   Curl::Setup(easy);
   const Net::ProgressAdapter progress_adapter{easy, progress};
-  easy.SetFailOnError();
+  // easy.SetFailOnError disabled: HTTP errors are dealt with here at the end
 
-  const auto mime = MakeUploadFlightMime(easy.Get(), settings,
+  const auto mime = MakeUploadFlightMime(easy.Get(), pilot,
                                          glider_type, igc_path);
   easy.SetMimePost(mime.get());
 
   Json::ParserOutputStream parser;
   const auto response =
     co_await Curl::CoStreamRequest(curl, std::move(easy), parser);
-  co_return parser.Finish();
+  auto json =  parser.Finish();
+  std::string error_msg = "";
```

Comment:
The declaration doesn't belong here; this variable is only used in certain code paths. Why is it a `std::string`, anyway? This adds a lot of overhead for copying string literals to a new heap allocation. Why does this variable exist at all? Just throw without a variable.

Reply:
I will rework this code part together with your backend and frontend discussion, so I want to path through this http code to the frontend function and do it handle there - without any exception here and also with any std::string stuff.